// @googlemaps/react-native-navigation-sdk/lib/commonjs/navigation/navigationView/navigationViewController.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getNavigationViewController = void 0;
var _viewManager = require("../../shared/viewManager");
/**
 * Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const getNavigationViewController = viewId => {
  return {
    setNavigationUIEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setNavigationUIEnabled, [isOn]);
    },
    setTripProgressBarEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setTripProgressBarEnabled, [isOn]);
    },
    setReportIncidentButtonEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setReportIncidentButtonEnabled, [isOn]);
    },
    setSpeedometerEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setSpeedometerEnabled, [isOn]);
    },
    setSpeedLimitIconEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setSpeedLimitIconEnabled, [isOn]);
    },
    setTrafficIncidentCardsEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setTrafficIncidentCardsEnabled, [isOn]);
    },
    setHeaderEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setHeaderEnabled, [isOn]);
    },
    setFooterEnabled: isOn => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setFooterEnabled, [isOn]);
    },
    showRouteOverview: () => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.showRouteOverview, []);
    },
    setNightMode: index => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setNightMode, [index]);
    },
    setRecenterButtonEnabled(isEnabled) {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setRecenterButtonEnabled, [isEnabled]);
    },
    setFollowingPerspective: perspective => {
      (0, _viewManager.sendCommand)(viewId, _viewManager.commands.setFollowingPerspective, [perspective]);
    }
  };
};
exports.getNavigationViewController = getNavigationViewController;
