// @googlemaps/react-native-navigation-sdk/lib/commonjs/navigation/navigation/useNavigationController.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useNavigationController = void 0;
var _reactNative = require("react-native");
var _react = require("react");
var _shared = require("../../shared");
var _types = require("./types");
var _navigationView = require("../navigationView");
/**
 * Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const {
  NavModule
} = _reactNative.NativeModules;
const useNavigationController = (termsAndConditionsDialogOptions, taskRemovedBehavior = _types.TaskRemovedBehavior.CONTINUE_SERVICE) => {
  const eventTransformer = (eventKey, ...args) => {
    if (eventKey === 'onRouteStatusResult' && typeof args[0] === 'string') {
      return [(0, _navigationView.getRouteStatusFromStringValue)(args[0])];
    }
    return args;
  };
  const moduleListenersHandler = (0, _shared.useModuleListeners)('NavEventDispatcher', ['onStartGuidance', 'onArrival', 'onLocationChanged', 'onNavigationReady', 'onRouteStatusResult', 'onRawLocationChanged', 'onRouteChanged', 'onReroutingRequestedByOffRoute', 'onTrafficUpdated', 'onRemainingTimeOrDistanceChanged', 'onNavigationInitError', 'onTurnByTurn', 'logDebugInfo'], eventTransformer);
  const navigationController = (0, _react.useMemo)(() => ({
    init: async () => {
      return await NavModule.initializeNavigator(termsAndConditionsDialogOptions, taskRemovedBehavior);
    },
    cleanup: async () => {
      moduleListenersHandler.removeAllListeners();
      await NavModule.cleanup();
    },
    setDestination: async (waypoint, routingOptions, displayOptions) => {
      return await NavModule.setDestination(waypoint, routingOptions, displayOptions);
    },
    setDestinations: async (waypoints, routingOptions, displayOptions) => {
      return await NavModule.setDestinations(waypoints, routingOptions, displayOptions);
    },
    continueToNextDestination: async () => {
      return await NavModule.continueToNextDestination();
    },
    clearDestinations: async () => {
      return await NavModule.clearDestinations();
    },
    startGuidance: async () => {
      return await NavModule.startGuidance();
    },
    stopGuidance: async () => {
      return await NavModule.stopGuidance();
    },
    setSpeedAlertOptions: async alertOptions => {
      return await NavModule.setSpeedAlertOptions(alertOptions);
    },
    setAbnormalTerminatingReportingEnabled: enabled => {
      return NavModule.setAbnormalTerminatingReportingEnabled(enabled);
    },
    setAudioGuidanceType: index => {
      NavModule.setAudioGuidanceType(index);
    },
    setBackgroundLocationUpdatesEnabled: isEnabled => {
      if (_reactNative.Platform.OS === 'ios') {
        NavModule.setBackgroundLocationUpdatesEnabled(isEnabled);
      }
    },
    setTurnByTurnLoggingEnabled: isEnabled => {
      NavModule.setTurnByTurnLoggingEnabled(isEnabled);
    },
    areTermsAccepted: async () => {
      return await NavModule.areTermsAccepted();
    },
    getCurrentRouteSegment: async () => {
      return await NavModule.getCurrentRouteSegment();
    },
    getRouteSegments: async () => {
      return await NavModule.getRouteSegments();
    },
    getCurrentTimeAndDistance: async () => {
      return await NavModule.getCurrentTimeAndDistance();
    },
    getTraveledPath: async () => {
      return await NavModule.getTraveledPath();
    },
    getNavSDKVersion: async () => {
      return await NavModule.getNavSDKVersion();
    },
    stopUpdatingLocation: () => {
      NavModule.stopUpdatingLocation();
    },
    startUpdatingLocation: () => {
      NavModule.startUpdatingLocation();
    },
    simulator: {
      simulateLocation: location => {
        NavModule.simulateLocation(location);
      },
      resumeLocationSimulation: () => {
        NavModule.resumeLocationSimulation();
      },
      pauseLocationSimulation: () => {
        NavModule.pauseLocationSimulation();
      },
      simulateLocationsAlongExistingRoute: ({
        speedMultiplier
      }) => {
        NavModule.simulateLocationsAlongExistingRoute(speedMultiplier);
      },
      stopLocationSimulation: () => {
        NavModule.stopLocationSimulation();
      }
    }
  }), [moduleListenersHandler, taskRemovedBehavior, termsAndConditionsDialogOptions]);
  return {
    navigationController,
    ...moduleListenersHandler
  };
};
exports.useNavigationController = useNavigationController;
